javacutil/src/main/java/org/checkerframework/javacutil/AbstractTypeProcessor.java

```diff
@@ -139,11 +139,20 @@ public void typeProcessingStart() {}
      * <p>Subclasses may override this method to do any aggregate analysis (e.g. generate report,
      * persistence) or resource deallocation.
      *
-     * <p>If an error (a Java error or a processor error) is reported, this method is not guaranteed
-     * to be invoked.
+     * @param hasError true if compilation issued an error, either from the Java compiler or from a
+     *     pluggable type-checker
      */
     public void typeProcessingOver() {}
 
+    /**
+     * Return log.
```

Comment:
Could you make this more descriptive?  A "log" is customarily unstructured text, but this has structure as explained at the com.sun.tools.javac.util.Log javadoc.